shared/src/main/scala/bloop/io/Environment.scala

```diff
@@ -0,0 +1,172 @@
+package bloop.io
+
+object Environment {
+
+  /**
+   * Returns "\n" if both OSTYPE and SHELL are nonEmpty, returns `System.lineSeparator` otherwise.
+   * @see [[shellLineSeparator]] which is similar.
+   */
+  def lineSeparator: String = if (validOstype.nonEmpty) shellLineSeparator else System.lineSeparator
+
+  /**
+   * Returns preferred SHELL environment lineSeparator:
+   *    '\r\n' on Windows, unless SHELL environment is recognized.
+   *    '\n'   everywhere else
+   *
+   * The default ending in Windows is '\r\n', but either of two conditions can
+   * change the value of [[shellLineSeparator]]:
+   *    1. if [[validShell]] is nonEmpty
+   *    2. if system property `"line.separator"` is modified
+   *
+   * Unlike [[lineSeparator]], only depends on the value of SHELL.
+   *
+   * @return preferred lineSeparator for SHELL environments, or `System.lineSeparator` otherwise.
+   */
+  private def shellLineSeparator: String = if (validShell.nonEmpty) "\n" else sysLineSeparator
+
+  /**
+   * Preferred line ending for SHELL environments.
+   *
+   * An alias for `shellLineSeparator`.
+   *
+   * @example
+   * {{{
+   *   print(someText + EOL)
+   * }}}
+   */
+  lazy val EOL: String = shellLineSeparator
+
+  /**
+   * Return the canonical lowercased shell name, or empty string.
+   *
+   * `shell` is valid if when lowercased, it has a `validShells` entry as substring.
+   *
+   * @return the matching `validShells` entry, if there is a match; empty string otherwise.
+   *
+   * @example if {{{SHELL=/usr/bin/bash}}}, returns {{{/bin/bash}}}.
+   *
+   */
+  private def validShell: String = validShells.find { _.endsWith(shellLC) } match {
+    case Some(sh) => sh
+    case _ => ""
+  }
+
+  /*
+   * Return lowercase shell with possible .exe extension removed.
+   */
+  private[this] def shellLC = shell.toLowerCase(java.util.Locale.ENGLISH) match {
+    case sh if sh.endsWith(".exe") => sh.replaceFirst(".exe$", "")
+    case sh => sh
+  }
+
+  /*
+   * Return the name of the SHELL environment variable, or empty string.
+   *
+   * Normally set in CYGWIN, MinGW, msys, Git Bash shell environments.
+   */
+  lazy val shell: String = getenvOpt("SHELL").getOrElse("")
+
+  /*
+   * Returns current value of system property `"line.separator"`.
+   *
+   * System.lineSeparator may only be redefined at JVM startup,
+   * whereas this may be redefined on the fly in client code.
+   *
+   * System.lineSeparator always returns the initial value of system property "line.separator".
+   * @see [[https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#lineSeparator--]]
+   *
+   * System properties may be redefined on JVM startup, e.g., from a shell environment:
+   * @example
+   *  {{{
+   *  $ java -J-Dline.separator=$'\n' ...
+   *  }}}
+   *
+   */
+  def sysLineSeparator: String = propsOpt("line.separator").getOrElse("\n")
```

Comment:
```suggestion
  private def sysLineSeparator: String = propsOpt("line.separator").getOrElse("\n")
```
we should only use `lineSeparator` if we want to make sure that we address the current issues everwhere.